### PR TITLE
fix(Harvest): Some regressions which affected the 2FA flow

### PR DESCRIPTION
**Pull request by doubleface**
Keep watching connection flow when connector execution is finished
Give connector information to the flow in all the contexts to allow it to get the associated policies

### CODE REVIEW

File: packages/cozy-harvest-lib/src/models/ConnectionFlow.js [hunk 1]
```diff
@@ -135,9 +135,10 @@ export const createOrUpdateAccount = async ({
  * This should be the go to source of truth for the state of a Konnector Job.
  */
 export class ConnectionFlow {
-  constructor(client, trigger) {
+  constructor(client, trigger, konnector) {
     this.client = client
     this.trigger = trigger
+    this.konnector = konnector
```
Comment (y-lohse) [inline packages/cozy-harvest-lib/src/models/ConnectionFlow.js:141]: Where do we use `this.konnector`?

Comment (doubleface) [reply packages/cozy-harvest-lib/src/models/ConnectionFlow.js:141]: https://github.com/cozy/cozy-libs/blob/686036095b80a4204e6ab0bd50cfe13e9bcafbb4/packages/cozy-harvest-lib/src/models/ConnectionFlow.js#L533

File: packages/cozy-harvest-lib/src/components/FlowProvider.jsx [hunk 2]
```diff
@@ -170,6 +170,10 @@ FlowProvider.propTypes = {
    * CozyClient instance
    */
   client: PropTypes.object.isRequired,
+  /**
+   * konnector manifest
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/FlowProvider.jsx:172]:
```suggestion
   * Konnector manifest
```

File: packages/cozy-harvest-lib/test/components/__snapshots__/TriggerManager.spec.js.snap [hunk 3]
```diff
@@ -123,6 +123,7 @@ exports[`TriggerManager when given an oauth konnector should redirect to OAuthFo
       "handleAccountUpdated": [Function],
       "handleJobUpdated": [Function],
       "handleTwoFA": [Function],
+      "konnector": undefined,
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/test/components/__snapshots__/TriggerManager.spec.js.snap:126]: Is it normal ?

Comment (ptbrowne) [reply packages/cozy-harvest-lib/test/components/__snapshots__/TriggerManager.spec.js.snap:126]: I see no tests testing the fix, could you add one please ?

Comment (doubleface) [reply packages/cozy-harvest-lib/test/components/__snapshots__/TriggerManager.spec.js.snap]: It was not normal, I fixed the unit tests and added one unit test to check that flow knows the konnector.

File: packages/cozy-harvest-lib/test/components/TriggerManager.spec.js [hunk 4]
```diff
@@ -66,7 +66,7 @@ const tMock = jest.fn()
 const client = new CozyClient({})
 const props = {
   konnector: fixtures.konnector,
-  flow: new ConnectionFlow(client),
+  flow: new ConnectionFlow(client, undefined, fixtures.konnector),
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/test/components/TriggerManager.spec.js:69]: Nit: At this point I am wondering if an option object would not be better, I had trouble remembering that the second argument is a trigger.

**Pull request merged**